### ISSUE TITLE
Conversion fail postgres->clickhouse

### ISSUE THREAD

**Issue by sudmed** [Altinity/clickhouse-sink-connector#235]
Hello, 
One of my postgres tables doesn't convert to Clickhouse (using the Lightweight version). In Clickhouse I see only new table with 0 rows:

<details><summary>table details from PG</summary>
<p>

```
            Column             |            Type             | Collation | Nullable |                       Default
-------------------------------+-----------------------------+-----------+----------+-----------------------------------------------------
 id                            | bigint                      |           | not null | nextval('person_id_seq'::text::regclass)
 lastname                      | text                        |           |          |
 middlename                    | text                        |           |          |
 firstname                     | text                        |           |          |
 birthday                      | date                        |           |          |
 sex                           | boolean                     |           |          |
 language                      | text                        |           |          |
 email                         | text                        |           |          |
 city                          | text                        |           |          |
 country                       | text                        |           |          |
 delete_date                   | timestamp(0) with time zone |           |          |
 avatarurl                     | text                        |           |          |
 user_id                       | bigint                      |           | not null | 0
 status                        | text                        |           |          | ''::text
 is_saved                      | boolean                     |           | not null | false
 is_main                       | boolean                     |           | not null | false
 file_token                    | text                        |           |          |
 system_token                  | text                        |           |          |
 is_online                     | boolean                     |           | not null | false
 online_status_id              | numeric(1,0)                |           |          |
 about                         | text                        |           |          |
 folder                        | text                        |           |          |
 phone_number                  | text                        |           |          |
 phone_number_confirmed        | boolean                     |           |          | false
 phone_number_code             | text                        |           |          |
 portal_id                     | bigint                      |           |          |
 TimeZone                      | text                        |           |          |
 is_phone_number_visible       | boolean                     |           | not null | false
 snils                         | text                        |           |          |
 inn                           | text                        |           |          |
 esia_user_id                  | bigint                      |           |          |
 region_id                     | bigint                      |           |          |
 was_deleted                   | boolean                     |           | not null | false
 template_id                   | bigint                      |           |          |
 parent_id                     | bigint                      |           |          |
 settings_can_be_changed       | boolean                     |           |          |
 is_vip                        | boolean                     |           |          |
 is_top                        | boolean                     |           |          |
 ic_id                         | uuid                        |           |          |
 time_zone                     | integer                     |           |          |
 external_id                   | text                        |           |          |
 comment                       | text                        |           |          |
 is_personalizated             | boolean                     |           | not null | false
 info_before                   | text                        |           |          |
 can_authorize                 | boolean                     |           | not null | false
 avatar_key                    | text                        |           |          |
 external_system_id            | uuid                        |           |          |
 system_id                     | uuid                        |           |          |
 address                       | text                        |           |          |
``` 

</p>
</details> 

<details><summary>show create table in CH</summary>
<p>

```
CREATE TABLE test_db.person
(
    `id` Int64,
    `lastname` Nullable(String),
    `middlename` Nullable(String),
    `firstname` Nullable(String),
    `birthday` Nullable(Date32),
    `sex` Nullable(Bool),
    `language` Nullable(String),
    `email` Nullable(String),
    `city` Nullable(String),
    `country` Nullable(String),
    `delete_date` Nullable(DateTime64(6)),
    `avatarurl` Nullable(String),
    `user_id` Int64,
    `status` Nullable(String),
    `is_saved` Bool,
    `is_main` Bool,
    `file_token` Nullable(String),
    `system_token` Nullable(String),
    `is_online` Bool,
    `online_status_id` Nullable(Decimal(1, 0)),
    `about` Nullable(String),
    `folder` Nullable(String),
    `phone_number` Nullable(String),
    `phone_number_confirmed` Nullable(Bool),
    `phone_number_code` Nullable(String),
    `portal_id` Nullable(Int64),
    `TimeZone` Nullable(String),
    `is_phone_number_visible` Bool,
    `snils` Nullable(String),
    `inn` Nullable(String),
    `esia_user_id` Nullable(Int64),
    `region_id` Nullable(Int64),
    `was_deleted` Bool,
    `template_id` Nullable(Int64),
    `parent_id` Nullable(Int64),
    `settings_can_be_changed` Nullable(Bool),
    `is_vip` Nullable(Bool),
    `is_top` Nullable(Bool),
    `ic_id` Nullable(UUID),
    `time_zone` Nullable(Int32),
    `external_id` Nullable(String),
    `comment` Nullable(String),
    `is_personalizated` Bool,
    `info_before` Nullable(String),
    `can_authorize` Bool,
    `avatar_key` Nullable(String),
    `external_system_id` Nullable(UUID),
    `system_id` Nullable(UUID),
    `address` Nullable(String),
    `_sign` Int8,
    `_version` UInt64
)
ENGINE = ReplacingMergeTree(_version)
ORDER BY tuple()

``` 

</p>
</details> 

And as @subkanthi said [here](https://github.com/Altinity/clickhouse-sink-connector/discussions/210#discussioncomment-5664783) I've changing `tuple()` to primary key `id`. The result is the same. Other tables were created and work well with @subkanthi's trick. 
Any idea which field breaks the conversion or why the table isn't parsed?

**Comment by subkanthi** [comment]
Hi @sudmed , can you please share the CREATE TABLE SQL in postgres along with the insert statements you are trying in postgres, so that its easier to reproduce, thanks.

**Comment by sudmed** [comment]
<details><summary>CREATE TABLE</summary>
<p>

```
CREATE TABLE person (
    id bigint DEFAULT nextval(('person_id_seq'::text)::regclass) NOT NULL,
    lastname text,
    middlename text,
    firstname text,
    birthday date,
    sex boolean,
    language text,
    email text,
    city text,
    country text,
    delete_date timestamp(0) with time zone,
    avatarurl text,
    user_id bigint DEFAULT 0 NOT NULL,
    status text DEFAULT ''::text,
    is_saved boolean DEFAULT false NOT NULL,
    is_main boolean DEFAULT false NOT NULL,
    file_token text,
    system_token text,
    is_online boolean DEFAULT false NOT NULL,
    online_status_id numeric(1,0),
    about text,
    folder text,
    phone_number text,
    phone_number_confirmed boolean DEFAULT false,
    phone_number_code text,
    portal_id bigint,
    "TimeZone" text,
    is_phone_number_visible boolean DEFAULT false NOT NULL,
    snils text,
    inn text,
    esia_user_id bigint,
    region_id bigint,
    was_deleted boolean DEFAULT false NOT NULL,
    template_id bigint,
    parent_id bigint,
    settings_can_be_changed boolean,
    is_vip boolean,
    is_top boolean,
    ic_id uuid,
    time_zone integer,
    external_id text,
    comment text,
    is_personalizated boolean DEFAULT false NOT NULL,
    info_before text,
    can_authorize boolean DEFAULT false NOT NULL,
    avatar_key text,
    external_system_id uuid,
    system_id uuid,
    address text
);
``` 

</p>
</details>

**Comment by aadant** [comment]
I can not see any primary key there :-) it is required to replicate the table

**Comment by sudmed** [comment]
It exists
```
ALTER TABLE ONLY person
    ADD CONSTRAINT person_pkey PRIMARY KEY (id);
```

**Comment by subkanthi** [comment]
Hi @sudmed ,
there was one bug with reading the primary key from postgres and thats why the `ORDER BY` defaulted to `ORDER BY tuple()`
This was fixed in #267 .

**Comment by subkanthi** [comment]
Fixed in #267
